packages/http/src/resources/Oracle.ts

```diff
@@ -10,9 +10,27 @@ export default class Oracle {
   }
 
   async getCurrentPrice(): Promise<any> {
-    const url = `/oracle/prices/current`
-    const { data: { data: { price } } } = await this.client.get(url)
-    
-    return new Balance(price, CurrencyType.usd)
+    const url = '/oracle/prices/current'
+    const { data: { data: { price, block } } } = await this.client.get(url)
+
+    return new Balance(price, CurrencyType.usd, block)
+  }
+
+  async getPriceAtBlock(height: number): Promise<any> {
+    const url = `/oracle/prices/${height}`
+    const { data: { data: { price, block } } } = await this.client.get(url)
+
+    return new Balance(price, CurrencyType.usd, block)
+  }
+
+  async getPredictedPrice(): Promise<any> {
+    const url = '/oracle/predictions'
+    const response = await this.client.get(url)
+    if (!response.data.data || !response.data.data.length) {
+      return []
```

Comment:
to be consistent with the way the API behaves, returning an empty array.  Could be changed.

Reply:
I think that's fine to stick with how the API behaves